build.rs

```diff
@@ -33,17 +32,33 @@ fn main() {
                 None => "clang-16".into(),
```

Comment:
The latest stable version of CLANG is 17,

`curl https://apt.llvm.org/llvm.sh 2>/dev/null | grep "CURRENT_LLVM_STABLE"`

Maybe we can search for clang-17 and clang-16 in order.

Reply:
I don't believe we have tested and used clang 17 as much as we do for clang 16. It might be a dangerous thing to pull the trigger here. In fact, the official git repo for LLVMs only have tags for RC versions of LLVM 17: https://github.com/llvm/llvm-project/releases

None of the major package manager has caught up on LLVM 17:

https://formulae.brew.sh/formula/llvm#default
https://archlinux.org/packages/extra/x86_64/clang/
https://src.fedoraproject.org/rpms/llvm

To me I think this is due to the apt package maintainer of LLVM making the switch too early.